drivers/rtc/rtc_rv8263.c

```diff
@@ -0,0 +1,701 @@
+    const struct device *dev;
+#endif
+
+#if (CONFIG_RTC_ALARM || CONFIG_RTC_UPDATE) && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+    struct gpio_callback gpio_cb;
+#endif
+
+#if CONFIG_RTC_ALARM && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+    rtc_alarm_callback alarm_cb;
+    void *alarm_cb_data;
+    bool is_alarm_pending;
+    struct k_work alarm_work;
+#endif
+
+#if CONFIG_RTC_UPDATE && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+    rtc_update_callback update_cb;
+    void *update_cb_data;
+    struct k_work update_work;
+#endif
+};
+
+#if (CONFIG_RTC_ALARM || CONFIG_RTC_UPDATE) && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+static void rv8263c8_gpio_callback_handler(const struct device *p_port, struct gpio_callback *p_cb,
+                                           gpio_port_pins_t pins)
+{
+    ARG_UNUSED(pins);
+    ARG_UNUSED(p_port);
+
+    struct rv8263c8_data *data = CONTAINER_OF(p_cb, struct rv8263c8_data, gpio_cb);
+
+#if CONFIG_RTC_ALARM && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+    k_work_submit(&data->alarm_work);
+#endif
+
+#if CONFIG_RTC_UPDATE && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+    k_work_submit(&data->update_work);
+#endif
+}
+#endif
+
+#if CONFIG_RTC_ALARM && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+static void rv8263_process_alarm(const struct device *dev)
+{
+    uint8_t reg;
+    struct rv8263c8_data *data = dev->data;
+    const struct rv8263c8_config *config = dev->config;
+
+    (void)k_sem_take(&data->lock, K_FOREVER);
+    i2c_reg_read_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2, &reg);
+    k_sem_give(&data->lock);
+
+    // Exit when the alarm flag isn´t set.
+    if (!(reg & RV8263C8_BM_AF)) {
+        LOG_DBG("Alarm flag is not set");
+        return;
+    }
+
+    data->is_alarm_pending = true;
+
+    if (data->alarm_cb != NULL) {
+        LOG_DBG("Calling alarm callback");
+        data->alarm_cb(dev, 0, data->alarm_cb_data);
+    }
+}
+
+static void rv8263c8_alarm_worker(struct k_work *p_work)
+{
+    struct rv8263c8_data *data = CONTAINER_OF(p_work, struct rv8263c8_data, alarm_work);
+
+    LOG_DBG("Process alarm interrupt from worker");
+
+    rv8263_process_alarm(data->dev);
+}
+#endif
+
+#if CONFIG_RTC_UPDATE && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+static int rv8263c8_update_enable_timer(const struct device *dev)
+{
+    int err;
+    uint8_t regs;
+    struct rv8263c8_data *data = dev->data;
+    const struct rv8263c8_config *config = dev->config;
+
+    if (dev == NULL) {
+        return -EINVAL;
+    }
+
+    regs = 0;
+    (void)k_sem_take(&data->lock, K_FOREVER);
+    err = i2c_burst_write_dt(&config->i2c_bus, RV8263C8_REGISTER_TIMER_VALUE, &regs, sizeof(regs));
+    if (err < 0) {
+        goto rv8263c8_update_enable_timer_exit;
+    }
+
+    // Configure the countdown timer for 1 Hz and enable pulse interrupts.
+    regs = (0x02 << 3) | (0x01 << 2) | (0x01 << 1) | (0x01 << 0);
```

Comment:
Please use defines here :) the values themselves are rather opaque, split it into multiple lines as well if required:
```
regs = SOME_FLAG_0 |
       SOME_FLAG_4 |
       SOME_FLAG_5;

```